src/base64.rs

```diff
@@ -251,10 +251,16 @@ impl FromBase64 for [u8] {
         let mut buf: u32 = 0;
         let mut modulus = 0;
 
-        let mut it = self.iter().enumerate();
+        let it = self.iter().enumerate();
+        let mut tail = false;
         for (idx, &byte) in it {
```

Comment:
I think this can be turned into `for (idx, &byte) in it.by_ref() {` so that `it` is not consumed.